From ba58d7d4032182dacade788e51c52aad61c2c84e Mon Sep 17 00:00:00 2001
From: Munehisa Kamata <kamatam@amazon.com>
Date: Fri, 17 Mar 2017 10:12:32 +0000
Subject: drivers: introduce AMAZON_DRIVER_UPDATES

This provides a central place to maintain out-of-tree drivers.
Renamed from VENDOR_AMAZON because the name was no longer appropriate.

Signed-off-by: Munehisa Kamata <kamatam@amazon.com>
Reviewed-by: Cristian Gafton <gafton@amazon.com>
Reviewed-by: Guru Anbalagane <guruanb@amazon.com>

CR: https://cr.amazon.com/r/6834563/
---
 drivers/Kconfig         |  2 ++
 drivers/Makefile        |  3 +++
 drivers/amazon/Kconfig  | 15 +++++++++++++++
 drivers/amazon/Makefile |  3 +++
 4 files changed, 23 insertions(+)
 create mode 100644 drivers/amazon/Kconfig
 create mode 100644 drivers/amazon/Makefile

diff --git a/drivers/Kconfig b/drivers/Kconfig
index e1e2066..c714e07 100644
--- a/drivers/Kconfig
+++ b/drivers/Kconfig
@@ -202,4 +202,6 @@ source "drivers/hwtracing/intel_th/Kconfig"
 
 source "drivers/fpga/Kconfig"
 
+source "drivers/amazon/Kconfig"
+
 endmenu
diff --git a/drivers/Makefile b/drivers/Makefile
index 733bf0b..400a757 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -174,3 +174,6 @@ obj-$(CONFIG_STM)		+= hwtracing/stm/
 obj-$(CONFIG_ANDROID)		+= android/
 obj-$(CONFIG_NVMEM)		+= nvmem/
 obj-$(CONFIG_FPGA)		+= fpga/
+
+# Amazon driver updates
+obj-$(CONFIG_AMAZON_DRIVER_UPDATES)	+= amazon/
diff --git a/drivers/amazon/Kconfig b/drivers/amazon/Kconfig
new file mode 100644
index 0000000..1a018dd
--- /dev/null
+++ b/drivers/amazon/Kconfig
@@ -0,0 +1,15 @@
+#
+# Amazon driver updates configuration
+#
+
+config AMAZON_DRIVER_UPDATES
+       bool "Amazon Driver Updates"
+       default y
+       depends on PCI || EXPERIMENTAL
+       ---help---
+         Amazon driver updates includes out-of-tree drivers and/or modifeid
+	 versions of the drivers present in the stable kernel tree.
+
+if AMAZON_DRIVER_UPDATES
+
+endif # AMAZON_DRIVER_UPDATES
diff --git a/drivers/amazon/Makefile b/drivers/amazon/Makefile
new file mode 100644
index 0000000..6b4996d
--- /dev/null
+++ b/drivers/amazon/Makefile
@@ -0,0 +1,3 @@
+#
+# Amazon Driver Updates
+#
-- 
2.7.5

